overlord/snapstate/snapstate.go

```diff
@@ -195,3 +199,51 @@ func Deactivate(s *state.State, snap string) (*state.TaskSet, error) {
 
 	return state.NewTaskSet(t), nil
 }
+
+// Retrieval functions
+
+func retrieveInfo(name string, si *snap.SideInfo) (*snap.Info, error) {
+	// XXX: move some of this in snap as helper?
+	snapYamlFn := filepath.Join(snap.MinimalPlaceInfo(name, si.Revision).MountDir(), "meta", "snap.yaml")
```

Comment:
Can we please have `snap.MountDir(name, revision), and have `Info.MountDir` implemented in terms of it?  Most of uses of `MinimalPlaceInfo` can probably be ported to it.